### PR TITLE
Convert deps test to spec

### PR DESCRIPTION
**Pull request by joshliu**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew tests` with your changes locally?

-----

### CODE REVIEW

**Comment by reitermarkus** [comment]
Great to have some help with this. Could you take a look at https://github.com/Homebrew/brew/pull/2026/files, and rebase this onto https://github.com/reitermarkus/brew/tree/spec-analytics, so you can use the same style?

**Comment by joshliu** [comment]
The test should run fine once setup_test_formula is added to the spec integration testing helper functions, which I saw that you already did in #2104.

**Comment by reitermarkus** [comment]
@joshliu, FYI, I already converted all other integration tests, just need to wait on  https://github.com/Homebrew/brew/pull/2104 for them as well.

**Comment by reitermarkus** [comment]
If you want to give another one a shot, these aren't done yet:

- `audit_test.rb`
- `formula_installer_test.rb`
- `formula_validation_test.rb`
- `uninstall_test.rb`
- `update_report_test.rb`
- `versions_test.rb`

**Comment by MikeMcQuaid** [comment]
@reitermarkus Will leave this one to you to review and merge when 💚.
@joshliu Nice work and thanks for the help!

File: Library/Homebrew/test/deps_spec.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      .and be_a_success
+  end
+
+  it "outputs a dependency for a Formula that has one dependency" do
+    expect { brew "deps", "bar" }.to output("foo\n").to_stdout
+      .and not_to_output.to_stderr
+      .and be_a_success
+  end
+
+  it "outputs dependencies on separate lines for a Formula that has multiple dependencies" do
```
Comment (reitermarkus) [inline Library/Homebrew/test/deps_spec.rb]: This seems to be testing recursive dependencies, not (just) multiple.

File: Library/Homebrew/test/deps_spec.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    setup_test_formula "foo"
+    setup_test_formula "bar"
+    setup_test_formula "baz", <<-EOS.undent
+      url "https://example.com/baz-1.0"
+      depends_on "bar"
+    EOS
+  end
+
+  it "outputs no dependencies for a Formula that has no dependencies" do
+    expect { brew "deps", "foo" }.to output("").to_stdout
```
Comment (reitermarkus) [inline Library/Homebrew/test/deps_spec.rb]: Also move the `.to output` on a separate line for all of these.

File: Library/Homebrew/test/deps_spec.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      url "https://example.com/baz-1.0"
+      depends_on "bar"
+    EOS
+  end
+
+  it "outputs no dependencies for a Formula that has no dependencies" do
+    expect { brew "deps", "foo" }
+      .to output("").to_stdout
+      .and not_to_output.to_stderr
+      .and be_a_success
```
Comment (reitermarkus) [inline Library/Homebrew/test/deps_spec.rb]: Reorder this like
```ruby
.to be_a_success
.and not_to_output.to_stdout
.and not_to_output.to_stderr
```

**Comment by reitermarkus** [comment]
Thanks, @joshliu!

**Pull request merged**